packages/runtime/test/transport/relationshipTemplates.test.ts

```diff
@@ -208,6 +209,99 @@ describe("Template Tests", () => {
+                expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                password: "password"
+            });
+            expect(createResult).toBeSuccessful();
+            expect(createResult.value.password).toBe("password");
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "password"
+            });
+            expect(loadResult).toBeSuccessful();
+            expect(loadResult.value.password).toBe("password");
+        });
+
+        test("send and receive a password-protected template via a token", async () => {
+            const templateId = (
+                await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                    content: emptyRelationshipTemplateContent,
+                    expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                    password: "password"
+                })
+            ).value.id;
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createTokenForOwnTemplate({ templateId });
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "password"
+            });
+            expect(loadResult).toBeSuccessful();
+            expect(loadResult.value.password).toBe("password");
+        });
+
+        test("error when loading a template with a wrong password", async () => {
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                content: emptyRelationshipTemplateContent,
+                expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                password: "password"
+            });
+            expect(createResult).toBeSuccessful();
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "wrong-password"
+            });
+            expect(loadResult).toBeAnError(/.*/, "error.platform.inputCannotBeParsed");
+        });
+
+        test("validation error when creating a template with a too short PIN", async () => {
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                content: emptyRelationshipTemplateContent,
+                expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                password: "1"
+            });
+            expect(createResult).toBeAnError(
+                "Your chosen 'password' is a PIN \\(consists of numbers only\\) and PINs must be at least 2 and at most 12 digits long",
```

Comment:
what are these backslashes?

Reply:
escapes, since this will become a regex

Reply:
if it's a regex, please make it a regex.

and DONT resolve comments that are not resolved.

Reply:
While looking into ways to not have people be surprised that the message in this matcher is currently a regex, I noticed that for runtime errors the error code is included in the error message, which is not the case for transport/... errors. I think this should be unified.